### ISSUE TITLE
Device Tellur WiFi Aroma Diffuser will not be switched on when changing fan values

### ISSUE THREAD

**Issue by lageorgem** [make-all/tuya-local#936]
This form is for reporting bugs.  If you have encountered a bug, you should
generally have some log messages or device diagnostics to show something is
going wrong.  If your problem is more of a question or a request for help,
then [Discussions](https://github.com/make-all/tuya-local/discussions) is
probably a more suitable venue.

**Describe the bug**
Device kgczidn0awas0prr (Tellur WiFi Aroma Diffuser) has successfully loaded using [this config](https://github.com/make-all/tuya-local/blob/main/custom_components/tuya_local/devices/yym_805SW_aroma_nightlight.yaml), but it seems that controlling it via the fan entity doesn't work completely. 
Namely, I can switch between 'small' and 'large' state if the device is on, but when switching off nothing happens. Same thing for switching from off to on.
Interestingly, in the device info I do have a toggle that can turn it on and off, it just doesn't work together with the fan control.

**To Reproduce**
Steps to reproduce the behavior including the device the issue was observed with:
Install device kgczidn0awas0prr (Tellur WiFi Aroma Diffuser) and attempt to turn it on from the `fan.aroma_diffuser` entity.

**Expected behavior**
The device turns on and off from the fan entity.

**Additional context**

The DPS values seem to be mapped correctly, please see debug logs:

```
2023-07-27 22:46:42.573 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Loaded device config yym_805SW_aroma_nightlight.yaml
2023-07-27 22:46:46.795 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Loaded device config yym_805SW_aroma_nightlight.yaml
2023-07-27 22:46:46.795 DEBUG (MainThread) [custom_components.tuya_local.helpers.config] Adding fan for fan
2023-07-27 22:46:46.801 DEBUG (MainThread) [custom_components.tuya_local.device] Refreshing device state for Aroma diffuser
2023-07-27 22:46:46.801 INFO (MainThread) [custom_components.tuya_local.device] Setting protocol version for Aroma diffuser to 3.3
2023-07-27 22:46:46.983 DEBUG (SyncWorker_3) [custom_components.tuya_local.helpers.device_config] Aroma diffuser: Mapped dps 1 value from True to True
2023-07-27 22:46:46.984 DEBUG (SyncWorker_3) [custom_components.tuya_local.helpers.device_config] Aroma diffuser: Mapped dps 1 value from True to True
2023-07-27 22:46:46.984 DEBUG (SyncWorker_3) [custom_components.tuya_local.helpers.device_config] Aroma diffuser: Mapped dps 2 value from small to 50
2023-07-27 22:46:46.984 DEBUG (SyncWorker_3) [custom_components.tuya_local.helpers.device_config] Aroma diffuser: Mapped dps 2 value from small to 50
2023-07-27 22:46:46.985 DEBUG (SyncWorker_3) [custom_components.tuya_local.helpers.device_config] Considering conditional mappings
2023-07-27 22:46:46.985 DEBUG (SyncWorker_3) [custom_components.tuya_local.helpers.device_config] Overriding speed values [] with [100, 50]
2023-07-27 22:46:46.985 DEBUG (SyncWorker_3) [custom_components.tuya_local.helpers.device_config] speed values: [100, 50]
2023-07-27 22:46:46.985 DEBUG (SyncWorker_3) [custom_components.tuya_local.helpers.device_config] Considering conditional mappings
2023-07-27 22:46:46.985 DEBUG (SyncWorker_3) [custom_components.tuya_local.helpers.device_config] Overriding speed values [] with [100, 50]
2023-07-27 22:46:46.985 DEBUG (SyncWorker_3) [custom_components.tuya_local.helpers.device_config] speed values: [100, 50]
2023-07-27 22:46:46.986 DEBUG (SyncWorker_3) [custom_components.tuya_local.device] Aroma diffuser refreshed device state: {"devId": "58107061e868e7c462da", "dps": {"1": true, "2": "small", "3": "1", "5": true, "6": "colour", "8": "ffe9000036ffff"}}
2023-07-27 22:46:46.986 DEBUG (SyncWorker_3) [custom_components.tuya_local.device] new state (incl pending): {"updated_at": 1690487206.9836123, "1": true, "2": "small", "3": "1", "5": true, "6": "colour", "8": "ffe9000036ffff"}
2023-07-27 22:46:47.072 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Aroma diffuser: Mapped dps 1 value from True to True
2023-07-27 22:46:47.072 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Aroma diffuser: Mapped dps 1 value from True to True
2023-07-27 22:46:47.072 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Aroma diffuser: Mapped dps 2 value from small to 50
2023-07-27 22:46:47.072 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Aroma diffuser: Mapped dps 2 value from small to 50
2023-07-27 22:46:47.072 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Considering conditional mappings
2023-07-27 22:46:47.072 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Overriding speed values [] with [100, 50]
2023-07-27 22:46:47.073 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] speed values: [100, 50]
2023-07-27 22:46:47.073 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Considering conditional mappings
2023-07-27 22:46:47.073 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Overriding speed values [] with [100, 50]
2023-07-27 22:46:47.073 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] speed values: [100, 50]
2023-07-27 22:46:47.097 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Loaded device config yym_805SW_aroma_nightlight.yaml
2023-07-27 22:46:47.097 DEBUG (MainThread) [custom_components.tuya_local.helpers.config] Adding light for light_nightlight
2023-07-27 22:46:47.099 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] color_mode values: ['hs', 'Colorful']
2023-07-27 22:46:47.099 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] color_mode values: ['hs', 'Colorful']
2023-07-27 22:46:47.100 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] color_mode values: ['hs', 'Colorful']
2023-07-27 22:46:47.100 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] color_mode values: ['hs', 'Colorful']
2023-07-27 22:46:47.100 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] color_mode values: ['hs', 'Colorful']
2023-07-27 22:46:47.100 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] color_mode values: ['hs', 'Colorful']
2023-07-27 22:46:47.100 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] color_mode values: ['hs', 'Colorful']
2023-07-27 22:46:47.100 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] color_mode values: ['hs', 'Colorful']
2023-07-27 22:46:47.101 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] color_mode values: ['hs', 'Colorful']
2023-07-27 22:46:47.101 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] color_mode values: ['hs', 'Colorful']
2023-07-27 22:46:47.101 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Aroma diffuser: Mapped dps 6 value from colour to hs
2023-07-27 22:46:47.101 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Aroma diffuser: Mapped dps 6 value from colour to hs
2023-07-27 22:46:47.101 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] color_mode values: ['hs', 'Colorful']
2023-07-27 22:46:47.101 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] color_mode values: ['hs', 'Colorful']
2023-07-27 22:46:47.101 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Aroma diffuser: Mapped dps 6 value from colour to hs
2023-07-27 22:46:47.102 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Aroma diffuser: Mapped dps 6 value from colour to hs
2023-07-27 22:46:47.102 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] format of >BBBHBB found
2023-07-27 22:46:47.102 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] format of >BBBHBB found
2023-07-27 22:46:47.102 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] format of >BBBHBB found
2023-07-27 22:46:47.102 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Aroma diffuser: Mapped dps 6 value from colour to hs
2023-07-27 22:46:47.102 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Aroma diffuser: Mapped dps 6 value from colour to hs
2023-07-27 22:46:47.103 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] color_mode values: ['hs', 'Colorful']
2023-07-27 22:46:47.104 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] color_mode values: ['hs', 'Colorful']
2023-07-27 22:46:47.104 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] color_mode values: ['hs', 'Colorful']
2023-07-27 22:46:47.104 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] color_mode values: ['hs', 'Colorful']
2023-07-27 22:46:47.104 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] color_mode values: ['hs', 'Colorful']
2023-07-27 22:46:47.105 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] color_mode values: ['hs', 'Colorful']
2023-07-27 22:46:47.105 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Aroma diffuser: Mapped dps 6 value from colour to hs
2023-07-27 22:46:47.105 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Aroma diffuser: Mapped dps 6 value from colour to hs
2023-07-27 22:46:47.105 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] color_mode values: ['hs', 'Colorful']
2023-07-27 22:46:47.105 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] color_mode values: ['hs', 'Colorful']
2023-07-27 22:46:47.105 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Aroma diffuser: Mapped dps 6 value from colour to hs
2023-07-27 22:46:47.105 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Aroma diffuser: Mapped dps 6 value from colour to hs
2023-07-27 22:46:47.106 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] format of >BBBHBB found
2023-07-27 22:46:47.106 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] format of >BBBHBB found
2023-07-27 22:46:47.106 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] format of >BBBHBB found
2023-07-27 22:46:47.106 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Aroma diffuser: Mapped dps 6 value from colour to hs
2023-07-27 22:46:47.106 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Aroma diffuser: Mapped dps 6 value from colour to hs
2023-07-27 22:46:47.106 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] color_mode values: ['hs', 'Colorful']
2023-07-27 22:46:47.260 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Loaded device config yym_805SW_aroma_nightlight.yaml
2023-07-27 22:46:47.260 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] option values: ['Off', '1 hour', '3 hours', '6 hours']
2023-07-27 22:46:47.260 DEBUG (MainThread) [custom_components.tuya_local.helpers.config] Adding select for select_timer
2023-07-27 22:46:47.261 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] option values: ['Off', '1 hour', '3 hours', '6 hours']
2023-07-27 22:46:47.262 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] option values: ['Off', '1 hour', '3 hours', '6 hours']
2023-07-27 22:46:47.262 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Aroma diffuser: Mapped dps 3 value from 1 to 1 hour
2023-07-27 22:46:47.262 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Aroma diffuser: Mapped dps 3 value from 1 to 1 hour
2023-07-27 22:46:47.262 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Aroma diffuser: Mapped dps 3 value from 1 to 1 hour
2023-07-27 22:46:47.262 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Aroma diffuser: Mapped dps 3 value from 1 to 1 hour
2023-07-27 22:46:47.262 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] option values: ['Off', '1 hour', '3 hours', '6 hours']
2023-07-27 22:46:47.262 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Aroma diffuser: Mapped dps 3 value from 1 to 1 hour
2023-07-27 22:46:47.262 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Aroma diffuser: Mapped dps 3 value from 1 to 1 hour
2023-07-27 22:46:47.263 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] option values: ['Off', '1 hour', '3 hours', '6 hours']
2023-07-27 22:46:47.263 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Aroma diffuser: Mapped dps 3 value from 1 to 1 hour
2023-07-27 22:46:47.263 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Aroma diffuser: Mapped dps 3 value from 1 to 1 hour
2023-07-27 22:46:47.263 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Aroma diffuser: Mapped dps 3 value from 1 to 1 hour
2023-07-27 22:46:47.263 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Aroma diffuser: Mapped dps 3 value from 1 to 1 hour
2023-07-27 22:46:47.263 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] option values: ['Off', '1 hour', '3 hours', '6 hours']
2023-07-27 22:46:47.264 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Aroma diffuser: Mapped dps 3 value from 1 to 1 hour
2023-07-27 22:46:47.264 DEBUG (MainThread) [custom_components.tuya_local.helpers.device_config] Aroma diffuser: Mapped dps 3 value from 1 to 1 hour
```

If the bug involves a device, then please include device diagnostics from
Settings / Devices & Services / Tuya Local (your device) / "1 device"

```
{
  "home_assistant": {
    "installation_type": "Home Assistant Container",
    "version": "2023.7.2",
    "dev": false,
    "hassio": false,
    "virtualenv": false,
    "python_version": "3.11.4",
    "docker": true,
    "arch": "aarch64",
    "timezone": "Europe/Bucharest",
    "os_name": "Linux",
    "os_version": "6.1.21-v8+",
    "run_as_root": true
  },
  "custom_components": {
    "sonoff": {
      "version": "3.5.2",
      "requirements": [
        "pycryptodome>=3.6.6"
      ]
    },
    "hacs": {
      "version": "1.32.1",
      "requirements": [
        "aiogithubapi>=22.10.1"
      ]
    },
    "tuya_local": {
      "version": "2023.7.0",
      "requirements": [
        "pycryptodome~=3.18",
        "tinytuya==1.12.9"
      ]
    },
    "browser_mod": {
      "version": "2.3.0",
      "requirements": []
    },
    "scheduler": {
      "version": "v0.0.0",
      "requirements": []
    },
    "ui_lovelace_minimalist": {
      "version": "v1.3.6",
      "requirements": [
        "aiofiles==0.8.0",
        "aiogithubapi>=22.2.4"
      ]
    },
    "xiaomi_cloud_map_extractor": {
      "version": "v2.2.0",
      "requirements": [
        "pillow",
        "pybase64",
        "python-miio",
        "requests",
        "pycryptodome"
      ]
    }
  },
  "integration_manifest": {
    "domain": "tuya_local",
    "name": "Tuya Local",
    "codeowners": [
      "@make-all"
    ],
    "config_flow": true,
    "dependencies": [],
    "documentation": "https://github.com/make-all/tuya-local",
    "integration_type": "device",
    "iot_class": "local_push",
    "issue_tracker": "https://github.com/make-all/tuya-local/issues",
    "requirements": [
      "pycryptodome~=3.18",
      "tinytuya==1.12.9"
    ],
    "version": "2023.7.0",
    "is_built_in": false
  },
  "data": {
    "name": "Aroma diffuser",
    "type": "yym_805SW_aroma_nightlight",
    "device_id": "**REDACTED**",
    "device_cid": "",
    "local_key": "**REDACTED**",
    "host": "**REDACTED**",
    "protocol_version": "auto",
    "tinytuya_version": "1.12.9",
    "api_version_set": 3.3,
    "api_version_used": 3.3,
    "api_working": true,
    "status": {},
    "cached_state": {
      "updated_at": 1690487563.0206099,
      "1": false,
      "2": "small",
      "3": "1",
      "5": true,
      "6": "colour",
      "8": "ffe9000036ffff"
    },
    "pending_state": {},
    "connected": true,
    "force_dps": [],
    "home_assistant": {
      "name": "Aroma diffuser",
      "name_by_user": null,
      "disabled": false,
      "disabled_by": null,
      "entities": [
        {
          "disabled": false,
          "disabled_by": null,
          "entity_category": null,
          "device_class": null,
          "original_device_class": null,
          "icon": null,
          "unit_of_measurement": null,
          "state": {
            "entity_id": "light.aroma_diffuser_nightlight",
            "state": "on",
            "attributes": {
              "effect_list": [
                "Colorful"
              ],
              "supported_color_modes": [
                "hs"
              ],
              "color_mode": "hs",
              "brightness": 255,
              "hs_color": [
                54,
                100
              ],
              "rgb_color": [
                255,
                229,
                0
              ],
              "xy_color": [
                0.476,
                0.49
              ],
              "icon": "mdi:light-recessed",
              "friendly_name": "Aroma diffuser Nightlight",
              "supported_features": 4
            },
            "last_changed": "2023-07-27T19:46:47.103150+00:00",
            "last_updated": "2023-07-27T19:46:47.103150+00:00"
          }
        },
        {
          "disabled": false,
          "disabled_by": null,
          "entity_category": "config",
          "device_class": null,
          "original_device_class": null,
          "icon": null,
          "unit_of_measurement": null,
          "state": {
            "entity_id": "select.aroma_diffuser_timer",
            "state": "1 hour",
            "attributes": {
              "options": [
                "Off",
                "1 hour",
                "3 hours",
                "6 hours"
              ],
              "icon": "mdi:timer",
              "friendly_name": "Aroma diffuser Timer"
            },
            "last_changed": "2023-07-27T19:46:47.263043+00:00",
            "last_updated": "2023-07-27T19:46:47.263043+00:00"
          }
        },
        {
          "disabled": false,
          "disabled_by": null,
          "entity_category": null,
          "device_class": null,
          "original_device_class": null,
          "icon": null,
          "unit_of_measurement": null,
          "state": {
            "entity_id": "fan.aroma_diffuser",
            "state": "off",
            "attributes": {
              "preset_modes": [],
              "percentage": 50,
              "percentage_step": 50.0,
              "preset_mode": null,
              "icon": "mdi:scent-off",
              "friendly_name": "Aroma diffuser",
              "supported_features": 1
            },
            "last_changed": "2023-07-27T19:51:09.898861+00:00",
            "last_updated": "2023-07-27T19:51:09.898861+00:00"
          }
        }
      ]
    }
  }
}
```

Please check if there are messages from Tuya Local in the Home
Assistant log (Settings / System / Logs), and include those too.

<img width="547" alt="image" src="https://github.com/make-all/tuya-local/assets/43441428/d1b3063f-09a7-40d5-968b-a307025e96ba">
<img width="299" alt="image" src="https://github.com/make-all/tuya-local/assets/43441428/70d23bbb-2f46-4fd9-9aba-8550ec629200">

**Comment by blakadder** [comment]
Your device seems to have different dp's or uses different values in the dp

If you have an iot.tuya.com account, please go to "Cloud" -> "API
Explorer".  Under "Device Control", select the "Query Things Data Model"
function, check the server is set correctly, and enter your device ID.

Please paste the output here.

**Comment by lageorgem** [comment]
```json
{
	"result": {
		"model": {
			"modelId": "0000031akw",
			"services": [{
				"actions": [],
				"code": "",
				"description": "",
				"events": [],
				"name": "默认服务",
				"properties": [{
					"abilityId": 1,
					"accessMode": "rw",
					"code": "switch_spray",
					"description": "",
					"extensions": {
						"iconName": "icon-dp_power2",
						"attribute": "17"
					},
					"name": "喷雾开关",
					"typeSpec": {
						"type": "bool",
						"typeDefaultValue": false
					}
				}, {
					"abilityId": 2,
					"accessMode": "rw",
					"code": "mode",
					"description": "",
					"extensions": {
						"iconName": "icon-dp_box"
					},
					"name": "工作模式",
					"typeSpec": {
						"range": ["large", "small"],
						"type": "enum",
						"typeDefaultValue": "large"
					}
				}, {
					"abilityId": 3,
					"accessMode": "rw",
					"code": "countdown",
					"description": "",
					"extensions": {
						"iconName": "icon-dp_time2"
					},
					"name": "倒计时",
					"typeSpec": {
						"range": ["cancel", "1", "3", "6"],
						"type": "enum",
						"typeDefaultValue": "cancel"
					}
				}, {
					"abilityId": 4,
					"accessMode": "ro",
					"code": "countdown_left",
					"description": "",
					"extensions": {
						"iconName": "icon-dp_time2"
					},
					"name": "倒计时剩余时间",
					"typeSpec": {
						"max": 360,
						"min": 0,
						"scale": 0,
						"step": 1,
						"type": "value",
						"typeDefaultValue": 0,
						"unit": "min"
					}
				}, {
					"abilityId": 5,
					"accessMode": "rw",
					"code": "switch_led",
					"description": "",
					"extensions": {
						"iconName": "icon-dp_power2",
						"attribute": "17"
					},
					"name": "灯光",
					"typeSpec": {
						"type": "bool",
						"typeDefaultValue": false
					}
				}, {
					"abilityId": 6,
					"accessMode": "rw",
					"code": "work_mode",
					"description": "",
					"extensions": {
						"iconName": "icon-dp_direction"
					},
					"name": "灯光模式",
					"typeSpec": {
						"range": ["colour", "colourful1"],
						"type": "enum",
						"typeDefaultValue": "colour"
					}
				}, {
					"abilityId": 7,
					"accessMode": "rw",
					"code": "bright_value",
					"description": "",
					"extensions": {
						"iconName": "icon-dp_light2"
					},
					"name": "亮度值",
					"typeSpec": {
						"max": 255,
						"min": 0,
						"scale": 0,
						"step": 1,
						"type": "value",
						"typeDefaultValue": 0,
						"unit": ""
					}
				}, {
					"abilityId": 8,
					"accessMode": "rw",
					"code": "colour_data",
					"description": "",
					"extensions": {
						"iconName": "icon-yanse"
					},
					"name": "彩光值",
					"typeSpec": {
						"maxlen": 255,
						"type": "string",
						"typeDefaultValue": ""
					}
				}]
			}]
		}
	},
	"success": true,
	"t": 1690647967170,
	"tid": "9eb12d922e2c11ee94f2faf13dd4e5eb"
}
```

**Comment by make-all** [comment]
I think the log from Home Assistant will be more useful.  That documentation seems compatible with the YYM aroma diffuser that the original report was about, the only differences are that it is missing the fault code dp (which is marked optional in the config, so should be OK), and has brightness listed, which is not included in the config, but given that the light only supports color mode, it is likely that brightness is controlled through the hsv dp and the brightness dp is a redundant carry-over from the  standard Tuya light template.

**Comment by make-all** [comment]
But actually for the fan, it seems to be a known issue - the behaviour of fans changed when the new UI was introduced in HA.  There used to be an implicit off within HA when the speed was set to 0, but that no longer seems to be the case - the off button in the UI just tries to set the speed to 0.

**Comment by lageorgem** [comment]
That's understandable, I suspected it could be an UI issue, but I decided to raise this bug in the hopes it's not. 
I'm just starting to understand how your configuration works. Would it be possible to map the speed 0 property to the switch property of the fan? I've seen you use something called `value_redirect`, but I'm not sure if that also supports a transformation. maybe using a condition together with that?
